src/app/clusters/scenes-server/SceneHandlerImpl.cpp

```diff
@@ -16,10 +16,224 @@
  */
 
 #include <app/clusters/scenes-server/SceneHandlerImpl.h>
+#include <app/util/endpoint-config-api.h>
+#include <app/util/odd-sized-integers.h>
 
 namespace chip {
 namespace scenes {
 
+namespace {
+
+template <int ByteSize, bool IsSigned>
+using OddSizedInteger        = app::OddSizedInteger<ByteSize, IsSigned>;
+using ConcreteAttributePath  = app::ConcreteAttributePath;
+using AttributeValuePairType = app::Clusters::ScenesManagement::Structs::AttributeValuePairStruct::Type;
+
+/// ConvertDefaultValueToWorkingValue
+/// @brief Helper function to convert a byte array to a value of the given type.
+/// @param EmberAfDefaultAttributeValue & defaultValue
+/// @return Value converted to the given working type
+template <typename Type>
+typename app::NumericAttributeTraits<Type>::WorkingType
+ConvertDefaultValueToWorkingValue(const EmberAfDefaultAttributeValue & defaultValue)
+{
+    if (sizeof(typename app::NumericAttributeTraits<Type>::WorkingType) <= 2)
+    {
+        return static_cast<typename app::NumericAttributeTraits<Type>::WorkingType>(defaultValue.defaultValue);
+    }
+
+    typename app::NumericAttributeTraits<Type>::StorageType sValue;
+    memcpy(&sValue, defaultValue.ptrToDefaultValue, sizeof(sValue));
+    return app::NumericAttributeTraits<Type>::StorageToWorking(sValue);
+}
+
+/// IsOnlyOneValuePopulated
+/// @brief Helper function to verify if only one value is populated in a given AttributeValuePairType
+/// @param AttributeValuePairType & type AttributeValuePairType to verify
+/// @return bool true if only one value is populated, false otherwise
+bool IsOnlyOneValuePopulated(const AttributeValuePairType & type)
```

Comment:
```suggestion
bool IsExactlyOneValuePopulated(const AttributeValuePairType & type)
```
since this checks that it's not _fewer_ than 1 as well.